pan3d/ui/globe.py

```diff
@@ -0,0 +1,883 @@
+import math
+from pathlib import Path
+
+from trame.decorators import TrameApp, change
+from trame.widgets import html, vuetify3 as v3
+
+from pan3d import catalogs as pan3d_catalogs
+from pan3d.utils.constants import XYZ, SLICE_VARS
+from pan3d.utils.convert import max_str_length
+from pan3d.utils.presets import PRESETS
+
+from pan3d.ui.css import base, preview
+from pan3d.ui.collapsible import CollapsableSection
+
+
+class SummaryToolbar(v3.VCard):
+    def __init__(
+        self,
+        t_labels="t_labels",
+        slice_t="slice_t",
+        slice_t_max="slice_t_max",
+        color_by="color_by",
+        data_arrays="data_arrays",
+        max_time_width="max_time_width",
+        max_time_index_width="max_time_index_width",
+        **kwargs,
+    ):
+        super().__init__(
+            classes="summary-toolbar",
+            rounded="pill",
+            **kwargs,
+        )
+
+        # Activate CSS
+        self.server.enable_module(base)
+        self.server.enable_module(preview)
+
+        with self:
+            with v3.VToolbar(
+                classes="pl-2",
+                height=50,
+                elevation=1,
+                style="background: none;",
+            ):
+                v3.VIcon("mdi-clock-outline")
+                html.Pre(
+                    f"{{{{ {t_labels}[slice_t] }}}}",
+                    classes="mx-2 text-left",
+                    style=(f"`min-width: ${{ {max_time_width} }}rem;`",),
+                )
+                v3.VSlider(
+                    prepend_inner_icon="mdi-clock-outline",
+                    v_model=(slice_t, 0),
+                    min=0,
+                    max=(slice_t_max, 0),
+                    step=1,
+                    hide_details=True,
+                    density="compact",
+                    flat=True,
+                    variant="solo",
+                    classes="mx-2",
+                )
+                html.Div(
+                    f"{{{{ {slice_t} + 1 }}}}/{{{{ {slice_t_max} + 1 }}}}",
+                    classes="mx-2 text-right",
+                    style=(f"`min-width: ${{ {max_time_index_width} }}rem;`",),
+                )
+                v3.VSelect(
+                    placeholder="Color By",
+                    prepend_inner_icon="mdi-format-color-fill",
+                    v_model=(color_by, None),
+                    items=(data_arrays, []),
+                    clearable=True,
+                    hide_details=True,
+                    density="compact",
+                    flat=True,
+                    variant="solo",
+                    max_width=200,
+                )
+
+
+class DataOrigin(CollapsableSection):
```

Comment:
Is that the same as the preview? Can you import and use the one from preview instead?